src/Item/PluginItem/Item.js

```diff
@@ -128,4 +185,17 @@ Item.contextTypes = {
     d2: PropTypes.object,
 };
 
-export default Item;
+const mapStateToProps = (state, ownProps) => ({
+    itemFilter: fromItemFilter.sGetFromState(state),
+    visualization: sGetVisualization(
+        state,
+        pluginManager.extractFavorite(ownProps.item).id
+    ),
+});
+
+const mapDispatchToProps = dispatch => ({
```

Comment:
I think its possible to write this in shorthand:

connect(mapStateToProps, {acReceivedActiveVisualization})(Item);

Then, in the component:

this.props.acReceivedActiveVisualization(id, type, activeType);

Reply:
But then we give this presentational component a crazy api. Presentational components should be reusable ideally.

When I use a ui component and want to pass in a handler I would expect to pass in "onSelectX" or similar, not "acReceivedActiveVisualization".

We could of course use "onSelectVisualization" and still do the shorthand. I usually extract mapState and mapDispatch, but that's just for readability.